### PR TITLE
feat: [FW-29] add internationalization using i18n and next-translate

### PR DESCRIPTION
**Pull request by barroro**
Changes:
- Add i18n setup and next-translate setup
- Create the `locales` folder to save all copies
    - Add 3 languages available for now (ca, es, en) 
    - Add translations for the Home page and layout (nav, footer etc...)
- Add LanguageSelector component to change between the languages available
    - Add Select component from the Shadcn library
- Add tests for the Footer component

### CODE REVIEW

File: locales/es/home.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "heroSection": {
+        "button": "Empieza",
+        "subtitle": "Genera listas de música en Spotify basado en les canciones que están tocando en el tour actual, usando la API de Setlist.fm",
```
Comment (DanielMoraDC) [inline locales/es/home.json:4]: "basado? -> "basadas"?

Comment (DanielMoraDC) [inline locales/es/home.json:4]: "les canciones" -> "las canciones"

File: locales/ca/home.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      "button": "Començar",
+      "subtitle": "Genera llistes de música a Spotify basat en les cançons que estàn tocant en aquell tour, utilizant la API de Setlist.fm",
+      "title": "Generador de llistes de <0>Spotify</0> utilitzant una llista de grups musicals"
+   },
+   "howItWorksSection": {
+      "step1": {
+         "title": "Inicia sessió en el teu compte de Spotify"
+      },
+      "step2": {
+         "title": "Busca i selecciona els artistes o grups"
```
Comment (DanielMoraDC) [inline locales/ca/home.json:12]: I would say "artistes o grups" is a bit redundant. What about leaving "artists/artistes/artistas" in all languages?

File: locales/en/home.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+    "heroSection": {
+        "button": "Get Started",
+        "subtitle": "Generate Spotify playlists based on the songs that it's playing in the current tour, using the Setlist.fm API",
+        "title": "Spotify playlist <0>generator</0> using a list of music bands"
```
Comment (DanielMoraDC) [inline locales/en/home.json:5]: "using" -> "given"?

File: locales/en/home.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        "step2": {
+            "title": "Search and select the artists or groups"
+        },
+        "step3": {
+            "title": "Check and customize the suggested list"
+        },
+        "title": "How it works"
+    },
+    "meta": {
+        "description": "Generate Spotify playlists based on the songs that it's playing in the current tour, using the Setlist.fm API",
```
Comment (DanielMoraDC) [inline locales/en/home.json:20]: "songs that it's" -> "songs that are"?

File: locales/ca/home.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+   "heroSection": {
+      "button": "Començar",
+      "subtitle": "Genera llistes de música a Spotify basat en les cançons que estàn tocant en aquell tour, utilizant la API de Setlist.fm",
```
Comment (DanielMoraDC) [inline locales/ca/home.json:4]: cançons que estan tocant -> cançons que estan sonant?

File: src/components/ui/Select.tsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+const SelectTrigger = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Trigger>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Trigger> & {
+    className?: string
+  }
+>(({ className, children, ...props }, ref) => (
+  <SelectPrimitive.Trigger
+    ref={ref}
+    className={cn(
+      "flex h-10 w-full items-center justify-between rounded-md border border-slate-200 bg-white px-3 py-2 text-sm ring-offset-white placeholder:text-slate-500 focus:outline-none focus:ring-2 focus:ring-dark focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 dark:border-slate-800 dark:bg-slate-950 dark:ring-offset-slate-950 dark:placeholder:text-slate-400 dark:focus:ring-slate-300",
```
Comment (DanielMoraDC) [inline src/components/ui/Select.tsx:22]: Just a nitpick: these long class names make the whole thing difficult to read. Would it make sense to move those to constants?

File: locales/es/common.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{
+    "footer": {
+        "allRightsReserved": "Todos los derechos reservados",
+        "selectLanguage": "Selecciona un idioma"
+    },
+    "languages": {
+        "catalan": "Catalan",
```
Comment (DanielMoraDC) [inline locales/es/common.json:7]: Catalan -> Catalán

File: src/components/ui/Select.tsx (hunk 6 above)
Comment (barroro) [reply src/components/ui/Select.tsx]: Done! I'll add some recommendations in README file to see and work better with the TailwindCSS classes. 👌

File: locales/ca/home.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
+{
+   "heroSection": {
+      "button": "Començar",
+      "subtitle": "Genera llistes de música a Spotify basat en les cançons que estàn sonant en aquell tour, utilizant la API de Setlist.fm",
```
Comment (DanielMoraDC) [inline locales/ca/home.json:4]:
```suggestion
      "subtitle": "Genera llistes de música a Spotify basades en les cançons que estan sonant en aquell tour",
```

Added a couple of minor changes.

Also: I am thinking that probably we do not need to provide details on the internals of the application,at least in the home page. Maybe we can add those details in another section.

File: locales/en/home.json [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{
+    "heroSection": {
+        "button": "Get Started",
+        "subtitle": "Generate Spotify playlists based on the songs that it's playing in the current tour, using the Setlist.fm API",
```
Comment (DanielMoraDC) [inline locales/en/home.json:4]:
```suggestion
        "subtitle": "Generate Spotify playlists based on the songs being played in the current tour
```

File: locales/es/common.json [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    },
+    "nav": {
+        "aboutUs": "Sobre nosotros",
+        "copyAccessToken": "Copiar Access Token",
+        "getStarted": "Empieza",
+        "login": "Iniciar sesion en Spotify",
+        "logout": "Cerrar sesion",
+        "privacyPolicy": "Politica de privacidad",
+        "profile": "Perfil",
+        "termsOfService": "Terminos de servicio"
```
Comment (DanielMoraDC) [inline locales/es/common.json:19]:
```suggestion
        "termsOfService": "Términos de servicio"
```

File: locales/es/common.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+        "catalan": "Catalan",
+        "english": "Inglés",
+        "spanish": "Español"
+    },
+    "nav": {
+        "aboutUs": "Sobre nosotros",
+        "copyAccessToken": "Copiar Access Token",
+        "getStarted": "Empieza",
+        "login": "Iniciar sesion en Spotify",
+        "logout": "Cerrar sesion",
```
Comment (DanielMoraDC) [inline locales/es/common.json:16]:
```suggestion
        "logout": "Cerrar sesión",
```

File: locales/es/common.json (hunk 7 above)
Comment (DanielMoraDC) [inline locales/es/common.json:7]:
```suggestion
        "catalan": "Catalán",
```

File: locales/es/common.json [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+    "languages": {
+        "catalan": "Catalan",
+        "english": "Inglés",
+        "spanish": "Español"
+    },
+    "nav": {
+        "aboutUs": "Sobre nosotros",
+        "copyAccessToken": "Copiar Access Token",
+        "getStarted": "Empieza",
+        "login": "Iniciar sesion en Spotify",
```
Comment (DanielMoraDC) [inline locales/es/common.json:15]:
```suggestion
        "login": "Iniciar sesión en Spotify",
```

File: locales/es/common.json [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+        "logout": "Cerrar sesion",
+        "privacyPolicy": "Politica de privacidad",
```
Comment (DanielMoraDC) [inline locales/es/common.json:17]:
```suggestion
        "privacyPolicy": "Política de privacidad",
```

File: locales/es/home.json [hunk 14]
```diff
@@ -0,0 +1,24 @@
+{
+    "heroSection": {
+        "button": "Empieza",
+        "subtitle": "Genera listas de música en Spotify basadas en las canciones que están tocando en el tour actual, usando la API de Setlist.fm",
+        "title": "Generador de listas en <0>Spotify</0> usando una lista de grupos musicales"
+    },
+    "howItWorksSection": {
+        "step1": {
+            "title": "Inicia session en tu cuenta de Spotify"
```
Comment (DanielMoraDC) [inline locales/es/home.json:9]:
```suggestion
            "title": "Inicia sesión en tu cuenta de Spotify"
```

**Review by DanielMoraDC — APPROVED** [review]

**Pull request merged**